File: kotlinames-common/src/main/java/com/nagopy/android/kotlinames/property/nullable/KNullableStringPropertyName.java
package com.nagopy.android.kotlinames.property.nullable;

import com.nagopy.android.kotlinames.KNullablePropertyName;
import com.nagopy.android.kotlinames.KSortablePropertyName;

public class KNullableStringPropertyName implements KNullablePropertyName<String>, KSortablePropertyName<String> {

    private final String name;

    public KNullableStringPropertyName(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }
}

File: kotlinames/src/main/kotlin/com/nagopy/android/kotlinames/property/nullable/KNullableStringPropertyName.kt
package com.nagopy.android.kotlinames.property.nullable

import com.nagopy.android.kotlinames.KNullablePropertyName
import com.nagopy.android.kotlinames.KSortablePropertyName

class KNullableStringPropertyName(val name: String) : KNullablePropertyName<String>, KSortablePropertyName<String> {

    override fun name(): String = name

}
